packages/components/src/Badge/Badge.component.js

```diff
@@ -1,67 +1,67 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
-import i18n from 'i18next';
+import i18n, { init } from 'i18next';
 import { translate } from 'react-i18next';
 
 import I18N_DOMAIN_COMPONENTS from '../constants';
 import Action from '../Actions/Action';
 import theme from './Badge.scss';
 
-function badgeClasses(tcStyle, onDelete) {
-	return classNames({
-		[theme['tc-badge']]: true,
-		'tc-badge': true,
-		[theme['tc-badge-outline']]: tcStyle === 'outline',
-		'tc-badge-outline': tcStyle === 'outline',
-		[theme['tc-badge-solid']]: tcStyle === 'solid',
-		'tc-badge-solid': tcStyle === 'solid',
-		[theme['tc-badge-with-icon']]: onDelete,
-		'tc-badge-with-icon': onDelete,
-	});
-}
-
-function badgeLabelClasses(onDelete) {
-	return classNames({
-		[theme['tc-badge-label']]: true,
-		'tc-badge-label': true,
-		[theme['tc-badge-label-with-icon']]: onDelete,
-		'tc-badge-label-with-icon': onDelete,
-	});
-}
-
-function renderDeleteIcon(onDelete, t) {
+function renderDeleteIcon(onClick, disabled, t) {
 	return (
 		<Action
 			label={t('BADGE_DELETE', { defaultValue: 'delete' })}
 			hideLabel
-			onClick={onDelete}
+			onClick={onClick}
+			disabled={disabled}
 			icon="talend-cross"
-			className={`${theme['tc-badge-delete-icon']} tc-badge-delete-icon`}
+			className={classNames('tc-badge-delete-icon', theme['tc-badge-delete-icon'])}
 		/>
 	);
 }
 
-function Badge({ label, tcStyle, onDelete, t }) {
+function Badge({ label, category, onDelete, onSelect, selected, disabled, t }) {
+	function onDeleteClick(e, ...rest) {
+		e.stopPropagation();
+		onDelete(e, ...rest);
+	}
+
+	const badgeClasses = classNames(
+		'tc-badge', theme['tc-badge'],
+		selected && ['tc-badge-selected', theme['tc-badge-selected']],
+		disabled && ['tc-badge-disabled', theme['tc-badge-disabled']],
+	);
+	const labelClasses = classNames('tc-badge-label', theme['tc-badge-label']);
+	const categoryClasses = classNames('tc-badge-category', theme['tc-badge-category']);
+
+	// noinspection Eslint
 	return (
-		<div className={badgeClasses(tcStyle, onDelete)}>
-			<span className={badgeLabelClasses(onDelete)}>
+		<div className={badgeClasses} onClick={!disabled && onSelect}>
+			{category && <span className={categoryClasses}>
+				{category}
+			</span>}
+			<span className={labelClasses}>
 				{label}
 			</span>
-			{onDelete && renderDeleteIcon(onDelete, t)}
+			{onDelete && renderDeleteIcon(onDeleteClick, disabled, t)}
```

Comment:
if `button`, we should display actions next to the button

Reply:
and position: relative ? meh